### PR TITLE
SONARJAVA-1021 RSPEC-2786 Nested "enum"s should not be declared static

### PR DESCRIPTION
**Pull request by merciesa**

### CODE REVIEW

File: java-checks/src/main/java/org/sonar/java/checks/NestedEnumStaticCheck.java [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+
+  @Override
+  public List<Tree.Kind> nodesToVisit() {
+    return ImmutableList.of(Tree.Kind.ENUM);
+  }
+
+  @Override
+  public void visitNode(Tree tree) {
+    Symbol.TypeSymbol symbol = ((ClassTree) tree).symbol();
+    if (symbol.owner().isTypeSymbol() && symbol.isStatic()) {
```
Comment (benzonico) [inline java-checks/src/main/java/org/sonar/java/checks/NestedEnumStaticCheck.java]: This check should not rely on semantic !!!

Comment (merciesa) [inline java-checks/src/main/java/org/sonar/java/checks/NestedEnumStaticCheck.java]: switched to syntax tree

File: java-checks/src/test/files/checks/NestedEnumStatic.java [hunk 2]
```diff
@@ -0,0 +1,14 @@
+enum MyEnum { // Compliant
+}
+
+// should be rejected during compilation
```
Comment (benzonico) [inline java-checks/src/test/files/checks/NestedEnumStatic.java]: Should not be required to be tested but ok.

**Comment by benzonico** [comment]
LGTM, please open relevant issue on symbol table if nested enums are not declared implicitely static.

**Pull request closed without merging**